libpromises/item_lib.c

```diff
@@ -510,40 +510,76 @@ void InsertAfter(Item **filestart, Item *ptr, const char *string)
     ip->classes = NULL;
 }
 
+/**
+ *  Splits a string containing a separator like ':' into a linked list of
+ *  separate items,
+ *
+ *  @NOTE backslashes can be used to escape either the separator, or the
+ *        backslash itself, e.g. "\:" or "\\" (ofcourse proper C strings need
+ *        double backslash).
```

Comment:
Note that backslash cannot be used as separator.
This is unfortunate when you consider splitting a DOS-path to get a list of components.